apm-lambda-extension/apmproxy/receiver.go

```diff
@@ -131,7 +131,7 @@ func (c *Client) handleIntakeV2Events() func(w http.ResponseWriter, r *http.Requ
 		}
 
 		if len(r.URL.Query()["flushed"]) > 0 && r.URL.Query()["flushed"][0] == "true" {
-			c.AgentDoneSignal <- struct{}{}
+			close(c.done)
```

Comment:
This doesn't look right: wouldn't this potentially lead to closing the channel multiple times (and panicking) when there are multiple Lambda invocations for the same extension lifetime? The extension may outlive a single invocation, i.e. on a warm start. In each invocation the APM agent may send a `?flushed=true` request to the extension.

Reply:
Thanks for spotting this! I've updated the code